### PR TITLE
docs: tetragon troubleshoot improve

### PR DESCRIPTION
**Pull request by janvi01**
Improved doc around tetra bugtool, demo how to run it for all different installation k8s, container or systemd host installation.

Fixes https://github.com/cilium/tetragon/issues/1974 second tick box.

### CODE REVIEW

File: docs/content/en/docs/troubleshooting/_index.md [hunk 1]
```diff
@@ -32,34 +32,81 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+
+### Systemd Host Installation
+
+1. **Execute tetra bugtool with Elevated Permissions:**
+
+```bash
+sudo tetra bugtool
+```
+
+> Note: Replace <...> placeholders with the actual values applicable to your setup.
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:81]: You can use the note shortcode:
```suggestion
{{< note >}}
Replace <...> placeholders with the actual values applicable to your setup.
{{< /note >}}
```

File: docs/content/en/docs/troubleshooting/_index.md [hunk 2]
```diff
@@ -32,34 +32,81 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+- Network configuration
+- Kernel configuration
+- eBPF maps
+- Process traces (if tracing is enabled)
+
+## Running tetra bugtool Demonstrations
+
+### Kubernetes
+
+1. **Identify Tetragon Pod:**
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:47]: I would avoid the bold emphasize please on your list items.

File: docs/content/en/docs/troubleshooting/_index.md [hunk 3]
```diff
@@ -32,34 +32,81 @@ environment for debugging. The tool is meant to be used for debugging a single
 Tetragon agent node. Note that in the context of Kubernetes, the command needs
 to be run from inside the Tetragon Pod's container.
 
+**Key Information Collected by the Bugtool:**
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:35]: Similarly I would avoid using bold for full sentences like that.

File: docs/content/en/docs/troubleshooting/_index.md [hunk 4]
```diff
@@ -32,34 +32,81 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+
+## Running tetra bugtool Demonstrations
+
+### Kubernetes
+
+1. **Identify Tetragon Pod:**
+
+```bash
+kubectl get pods -n <tetragron-namespace> -l app.kubernetes.io/name=tetragon
+```
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:51]: You can shift by two your codeblocks so that they fit into the item indentation.
```suggestion
  ```bash
  kubectl get pods -n <tetragron-namespace> -l app.kubernetes.io/name=tetragon
  ```
```

File: docs/content/en/docs/troubleshooting/_index.md [hunk 5]
```diff
@@ -32,34 +32,81 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+
 ## Enable debug log level
 
 When debugging, it might be useful to change the log level. The default log
 level is controlled by the log-level option at startup:
 
-* Enable debug level with `--log-level=debug`
-* Enable trace level with `--log-level=trace`
+- Enable debug level with `--log-level=debug`
+- Enable trace level with `--log-level=trace`
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:89]: please avoid making unnecessary changes especially for git history, this doesn't modify anything when rendered.

File: docs/content/en/docs/troubleshooting/_index.md [hunk 6]
```diff
@@ -32,34 +32,81 @@ environment for debugging. The tool is meant to be used for debugging a single
...
   sudo kill -s SIGRTMIN+21 $(pidof tetragon)
   ```
 
-* To Restore the original log level send the `SIGRTMIN+22` signal to tetragon pid:
+- To Restore the original log level send the `SIGRTMIN+22` signal to tetragon pid:
 
   ```shell
   sudo kill -s SIGRTMIN+22 $(pidof tetragon)
   ```
-
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:65]: similarly with whitespace codechange

**Review by mtardy — CHANGES_REQUESTED** [review]
Thanks a lot for taking the time to do that, I have some remarks on the form, I hope it's not too much.

File: docs/content/en/docs/troubleshooting/_index.md (hunk 1 above)
Comment (janvi01) [reply docs/content/en/docs/troubleshooting/_index.md]: I wonder if it is working as it isn't reflected in the preview.

File: docs/content/en/docs/troubleshooting/_index.md (hunk 4 above)
Comment (janvi01) [reply docs/content/en/docs/troubleshooting/_index.md]: shifted by three whitespace to align with indentation

File: docs/content/en/docs/troubleshooting/_index.md (hunk 1 above)
Comment (mtardy) [reply docs/content/en/docs/troubleshooting/_index.md]: you should be able to see it here https://deploy-preview-2115--tetragon.netlify.app/docs/troubleshooting/ :)

File: docs/content/en/docs/troubleshooting/_index.md [hunk 7]
```diff
@@ -32,6 +32,56 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+
+   ```bash
+   docker exec -it <tetragron-container-id> /bin/sh
+   ```
+
+2. Run tetra bugtool:
+
+   ```bash
+   tetra bugtool
+   ```
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:71]: Can you directly execute `tetra bugtool` instead of `/bin/sh` and then `tetra bugtool`?

**Review by mtardy — COMMENTED** [review]
Thanks a lot, to me it looks good! Could you squash your commits into one with only your first commit message? Thanks :)

**Comment by mtardy** [comment]
Hey, I saw you added a merge commit. Please use `git rebase -i <your_first_branch_commit>^` then drop the merge commit and squash all the commits into the first one. Then you can just resync your main branch and when on your dev branch, just `git rebase main`.

File: docs/content/en/docs/troubleshooting/_index.md [hunk 8]
```diff
@@ -32,6 +32,56 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+- Process traces (if tracing is enabled)
+
+## Running tetra bugtool Demonstrations
+
+### Kubernetes
+
+1. Identify Tetragon Pod:
+
+   ```bash
+   kubectl get pods -n <tetragron-namespace> -l app.kubernetes.io name=tetragon
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:50]:
```suggestion
   kubectl get pods -n <tetragon-namespace> -l app.kubernetes.io name=tetragon
```

File: docs/content/en/docs/troubleshooting/_index.md [hunk 9]
```diff
@@ -32,6 +32,56 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+   ```
+
+2. Execute tetra bugtool within the Pod:
+
+   ```bash
+   kubectl exec -n <tetragron-namespace> <tetragron-pod-name> -- tetra bugtool
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:56]:
```suggestion
   kubectl exec -n <tetragon-namespace> <tetragron-pod-name> -- tetra bugtool
```

File: docs/content/en/docs/troubleshooting/_index.md [hunk 10]
```diff
@@ -32,6 +32,56 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+   ```
+
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:58]: Can you add the step to retrieve the created archive from the pods fs to your local system (using `kubectl cp` for example)

File: docs/content/en/docs/troubleshooting/_index.md [hunk 11]
```diff
@@ -32,6 +32,56 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+### Container Installation
+
+1. Enter the Tetragon Container:
+
+   ```bash
+   docker exec -it <tetragron-container-id>
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:64]:
```suggestion
   docker exec -it <tetragron-container-id> tetra bugtool
```

File: docs/content/en/docs/troubleshooting/_index.md [hunk 12]
```diff
@@ -32,6 +32,56 @@ environment for debugging. The tool is meant to be used for debugging a single
...
+   ```
+
+2. Run tetra bugtool:
+
+   ```bash
+   tetra bugtool
+   ```
```
Comment (mtardy) [inline docs/content/en/docs/troubleshooting/_index.md:71]: Can you replace this with the step to retrieve the created archive from the container fs to your local system (using docker cp for example)

**Review by mtardy — CHANGES_REQUESTED** [review]
All right I did the fix for you, here are a few things to consider please on the content this time, sorry I didn't do everything at once. Please try to amend the existing commit with your changes or add commits and then squash them.

**Comment by janvi01** [comment]
@mtardy I did the required changes and amended the commit (apologies quite confused, how did I get others commits here).

**Comment by mtardy** [comment]
> @mtardy I did the required changes and amended the commit (apologies quite confused, how did I get others commits here).

Thanks for taking the time to do the changes!

Sorry this time you'll need to cleanup your PR. Instead of opening a new PR like last time or me tidying up stuff, I think you should take some time to get more familiar with git and `git rebase` in particular, that will be really useful for you! :)

**Comment by janvi01** [comment]
@mtardy I squash the commits into a single one, but how to remove others' file changes. Apologies for the inconvenience.

**Review by mtardy — APPROVED** [review]
I'm going to merge that, I forced pushed again removing all the changes from other commits you merged into your initial commit.

While we really appreciate you taking the time to write documentation, please take a bit more time next time testing the commands you put in the documentation. Almost all of them were not working.
- the `kubectl get pods -l app.kubernetes.io/name=tetragon` label was wrong, with a space instead of the `/`
- the `kubectl exec` was missing the `-c tetragon` option
- the `kubectl cp` was missing the `-c tetragon` and the archive name was wrong.
- the `docker cp` had a wrong archive name.
